### DISCUSSION TITLE
Welcome to CoCo C Discussions!

### DISCUSSION THREAD

**Discussion by Deek** [Deek/CoCoC#11] [General]
## 👋 Welcome!
Discussions are a place to connect with other members of the OS-9/6x09 and CoCo C community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, please comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by allenhuffman** [comment]
I will want to get started fresh. My last C work was in t 90s using the OS-9 stuff with a few patches. Is there a good getting started showing NitrOS-9 and RS-DOS solutions?

**Reply by Deek** [reply]
> I will want to get started fresh. My last C work was in t 90s using the OS-9 stuff with a few patches. Is there a good getting started showing NitrOS-9 and RS-DOS solutions?

I don't know about Disk BASIC, but on the OS-9 front the language hasn't really changed a great deal. It's still using old-school "static" libraries, so the more library functions you use the more of your program is composed of those library functions.

The DCC compiler "ships" with the newer C library assembled by Carl Kreider, which has more granularity to it and is closer to the library shipped with OSK.

We also ship a version of Tim Kientzle's CGFX 7 library, though I have updated it with bug fixes and support for additional features from NitrOS-9 GrfDrv.

Our C preprocessor is, as far as I am aware, compatible with ANSI and ISO C90, but the compiler still doesn't support prototypes. That's a very hard problem to solve.

Recently I added support for `unsigned char` to the compiler, so chars with the high bit set can have values higher than 127.

**Reply by allenhuffman** [reply]
Sounds good. My dev system is still stuck in 1995 with the latest patches we had at the time.

**Reply by Deek** [reply]
The easiest way to switch to `dcc` is to use EOU and change the file `/DD/SYS/Make.default` to set Make's `CC` variable to read
```sh
CC=dcc
```
Then your Makefiles will automatically use `dcc` to build code.
